src/main/java/com/fasterxml/jackson/core/util/DefaultPrettyPrinter.java

```diff
@@ -255,7 +255,8 @@ protected DefaultPrettyPrinter _withSpaces(boolean state)
      */
     public DefaultPrettyPrinter withSeparators(Separators separators) {
         _separators = separators;
-        _objectFieldValueSeparatorWithSpaces = " " + separators.getObjectFieldValueSeparator() + " ";
+        _objectFieldValueSeparatorWithSpaces = separators.getObjectFieldValueSpacing().apply(
```

Comment:
For performance reasons, let's also add the other 2 "WithSpaces" fields that are missing to avoid calling `apply()` on those cases.

Reply:
Done